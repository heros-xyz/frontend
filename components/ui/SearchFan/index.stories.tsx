import type { Meta, StoryObj } from "@storybook/react";
import { ChangeEvent } from "react";
import SearchFan from ".";

const meta: Meta<typeof SearchFan> = {
  title: "Fan/SearchFan",
  component: SearchFan,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {},
};
export default meta;
type Story = StoryObj<typeof SearchFan>;

export const SearchFanAthletelist: Story = {
  render: (args) => <SearchFan {...args}>Search with name</SearchFan>,
  args: {
    placeholder: "Search Fans",
    handleChange: (e: ChangeEvent<HTMLInputElement>) => {
      console.log(e.target.value);
    },
  },
};
