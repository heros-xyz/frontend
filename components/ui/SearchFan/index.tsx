import {
  Flex,
  Input,
  Box,
  InputGroup,
  InputLeftElement,
  BoxProps,
} from "@chakra-ui/react";
import React from "react";
import { FindIcon } from "@/components/svg/Find";

interface SearchProp extends BoxProps {
  placeholder: string;
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const SearchFan: React.FC<SearchProp> = ({
  placeholder,
  handleChange,
  ...props
}) => {
  return (
    <Box {...props}>
      <Flex h={14} pt={{ base: "22px", lg: "30px" }}>
        <InputGroup>
          <InputLeftElement pointerEvents="none" w="18" h="18">
            <FindIcon color="primary" mt="18px" />
          </InputLeftElement>
          <Input
            type="text"
            placeholder={placeholder}
            _hover={{
              borderColor: "primary",
            }}
            _placeholder={{ color: "grey.100" }}
            _focusVisible={{ boxShadow: "none", outline: "none" }}
            color="primary"
            borderColor="primary"
            border={0}
            borderRadius={0}
            borderBottom="1px"
            onChange={handleChange}
            fontSize={{ base: "md", xl: "lg" }}
          />
        </InputGroup>
      </Flex>
    </Box>
  );
};

export default SearchFan;
