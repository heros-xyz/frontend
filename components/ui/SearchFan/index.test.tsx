import { render } from "@testing-library/react";
import { ChangeEvent } from "react";
import SearchFan from ".";

it("renders homepage unchanged", () => {
  const { container } = render(
    <SearchFan
      placeholder="Search Fans"
      handleChange={(e: ChangeEvent<HTMLInputElement>) => {
        console.log(e.target.value);
      }}
    />
  );
  expect(container).toMatchSnapshot();
});
