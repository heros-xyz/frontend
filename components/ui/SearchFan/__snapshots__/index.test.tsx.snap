// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-1fwkvq8"
    >
      <div
        class="chakra-input__group css-bx0blc"
      >
        <div
          class="chakra-input__left-element css-dyheo2"
        >
          <svg
            class="chakra-icon css-1n4zmxg"
            focusable="false"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.5 16.5L11.5 11.5M13.1667 7.33333C13.1667 10.555 10.555 13.1667 7.33333 13.1667C4.11167 13.1667 1.5 10.555 1.5 7.33333C1.5 4.11167 4.11167 1.5 7.33333 1.5C10.555 1.5 13.1667 4.11167 13.1667 7.33333Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </svg>
        </div>
        <input
          class="chakra-input css-1kfcnoc"
          placeholder="Search Fans"
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
